### PR TITLE
ARC 2068 - Handling SSO Login in UI

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- Identified the SSO login users who can not connect their orgs.
- Displaying information to those users to correctly connect their orgs.

**Why**
- SSO login users can not click on the `CONNECT` button and have no idea why.

**How has this been tested?**  
- Locally

**Screenshot**
image

### CODE REVIEW

File: src/routes/github/configuration/github-configuration-get.ts [hunk 1]
```diff
@@ -101,6 +101,7 @@ const getInstallationsWithAdmin = async (
 			...installation,
 			numberOfRepos,
 			isAdmin,
+			isSso: !!errors.find(err => err?.cause?.response?.headers["x-github-sso"]),
```
Comment (bgvozdev) [inline src/routes/github/configuration/github-configuration-get.ts:104]: It doesn't look too reliable tbh... Could we maybe modify isUserAdminOfOrganization to return some result object/enum that returns more details?

File: src/routes/github/github-oauth-router.ts [hunk 2]
```diff
@@ -141,6 +141,9 @@ export const GithubAuthMiddleware = async (req: Request, res: Response, next: Ne
 	try {
 		const { query, originalUrl } = req;
 		if (query && query["resetGithubToken"]) {
+			req.session.githubToken = undefined;
+			req.session.githubRefreshToken = undefined;
```
Comment (bgvozdev) [inline src/routes/github/github-oauth-router.ts:145]: Curious why the tests were not failing? I think we should them test cause they don't look reliable...

File: src/routes/github/configuration/github-configuration-get.ts (hunk 1 above)
Comment (krazziekay) [reply src/routes/github/configuration/github-configuration-get.ts]: Created a new Error instance `GithubClientSSOLoginError` and matched the type of error with this type

File: src/util/github-utils.ts [hunk 3]
```diff
@@ -14,6 +14,8 @@ export const isUserAdminOfOrganization = async (githubClient: GitHubUserClient,
 		return role === "admin";
 	} catch (err) {
 		logger.warn({ err, orgName, username }, `could not determine admin status of user in org`);
+		err.requiresSsoLogin = err?.cause?.response?.headers["x-github-sso"];
```
Comment (bgvozdev) [inline src/util/github-utils.ts:17]: Shouldn't error already be an instance of GithubClientSSOLoginError ? Why do we need this flag?

I think in github client interceptors we should check for:
 - status (=401?) 
 - the header

If these match, we rethrow the error as GithubClientSSOLoginError and then just use instanceof everywhere. Wdyt?

Comment (krazziekay) [reply src/util/github-utils.ts]: Updated as suggested.

**Review by bgvozdev — APPROVED** [review]

File: src/routes/github/github-oauth-router.ts (hunk 2 above)
Comment (krazziekay) [reply src/routes/github/github-oauth-router.ts:145]: Test case added

**Review by bgvozdev — APPROVED** [review]

**Pull request merged**